src/model/element.ts

```diff
@@ -32,13 +33,14 @@ export class ScannedElement implements Resolvable {
   className?: string;
   superClass?: string;
   extends?: string;
-  properties: ScannedProperty[] = [];
-  attributes: ScannedAttribute[] = [];
+  properties = [] as ScannedProperty[];
```

Comment:
I like that you're experimenting! :)

I personally prefer the first form (matches non default property syntax & uses types to describe the property instead of the value being set to that property) but I guess that's subjective. I'm more concerned with moving away from the format we've been using everywhere else so far and would prefer we stick to that.